cmake/onnxruntime_mlas.cmake

```diff
@@ -93,6 +93,35 @@ else()
       ${ONNXRUNTIME_ROOT}/core/mlas/lib/aarch64/sgemma.s
     )
 
+  elseif (CMAKE_SYSTEM_PROCESSOR MATCHES "^(i.86|x86?)$")
```

Comment:
Could you add a few comments here, what kind of string it want to match? Please list a few examples here.

Reply:
Come on, I picked this up from a cmake builtin module to detect x86 to impress you! The only one that has been tested is x86 though, so this could be changed back to the style used for x86_64 if preferred.